test/solution-template-validation-tests/test/mainTemplateTests.js

```diff
@@ -82,22 +82,16 @@ describe('template files - ', () => {
 
         });
 
-        // Not required pending Jianping's PR
-        it('a parameter named "location" must exist and must be used on all resource "location" properties', () => {
+        // The location parameter MUST NOT have allowedValues property and MUST have defaultValue property whose value MUST be '[resourceGroup().location]'
+        it('a parameter named "location" must exist and it must have a defaultValue of resourceGroup().location', () => {
             mainTemplateFileJSONObject.should.withMessage('file:mainTemplate.json is missing parameters property').have.property('parameters');
             mainTemplateFileJSONObject.parameters.should.withMessage('file:mainTemplate.json is missing location property in parameters').have.property('location');
 
-            // TODO: if location default value exists, it should be set to resourceGroup.location(). Correct?
-
-            // each resource location should be "location": "[parameters('location')]"
-            var expectedLocation = '[parameters(\'location\')]';
-            var message = 'in file:mainTemplate.json should have location set to [parameters(\'location\')]';
-            Object.keys(mainTemplateFileJSONObject.resources).forEach(resource => {
-                var val = mainTemplateFileJSONObject.resources[resource];
-                if (val.location && val.type.toLowerCase() != 'microsoft.resources/deployments') {
-                    val.location.split(' ').join('').should.withMessage(getErrorMessage(val, mainTemplateFile, message)).be.eql(expectedLocation);
-                }
-            });
+            // The location parameter must have a defaultValue property, and its value must be '[resourceGroup().location]'
+            var location = mainTemplateFileJSONObject.parameters.location;
+            location.should.withMessage('file:mainTemplate.json location property must have a default value').have.property('defaultValue');
```

Comment:
in file:mainTemplate.json, location property MUST have defaultValue

Reply:
corrected